components/brave_new_tab_ui/components/default/rewards/index.tsx

```diff
@@ -0,0 +1,274 @@
+    } = this.props
+
+    if (walletCreateFailed || walletCorrupted) {
+      return getLocale('walletFailedButton')
+    }
+
+    if (walletCreating) {
+      return getLocale('rewardsWidgetTurningOn')
+    }
+
+    return getLocale('rewardsWidgetTurnOnRewards')
+  }
+
+  optInAction = (hasEnabled: boolean) => {
+    if (hasEnabled) {
+      this.props.onEnableRewards()
+    } else {
+      this.props.onCreateWallet()
+    }
+  }
+
+  renderPreOptIn = () => {
+    const {
+      enabledMain,
+      walletCreated
+    } = this.props
+
+    if (enabledMain && walletCreated) {
+      return null
+    }
+
+    const hasEnabled = !enabledMain && walletCreated
+
+    return (
+      <>
+        <PreOptInInfo>
+          <Title>
+            {
+              hasEnabled
+              ? getLocale('rewardsWidgetReEnableTitle')
+              : getLocale('rewardsWidgetEnableTitle')
+            }
+          </Title>
+          <SubTitle>
+            {
+              hasEnabled
+              ? getLocale('rewardsWidgetReEnableSubTitle')
+              : getLocale('rewardsWidgetEnableSubTitle')
+            }
+          </SubTitle>
+        </PreOptInInfo>
+        <PreOptInAction>
+          <TurnOnButton
+            data-test-id={'optInAction'}
+            onClick={this.optInAction.bind(this, hasEnabled)}
+          >
+            {this.getButtonText()}
+          </TurnOnButton>
+        </PreOptInAction>
+      </>
+    )
+  }
+
+  renderAmountItem = (type: string) => {
+    const {
+      balance,
+      enabledAds,
+      onEnableAds,
+      adsEstimatedEarnings
+    } = this.props
+
+    const rates = balance.rates || {}
+    const showEnableAds = type === 'ads' && !enabledAds
+    const amount = type === 'ads'
+      ? adsEstimatedEarnings.toFixed(1)
+      : this.getTotalContributions()
+    const converted = convertBalance(amount, rates)
+
+    return (
+      <AmountItem>
+        {
+          showEnableAds
+          ? <TurnOnAdsButton onClick={onEnableAds}>
+              {this.getButtonText(true)}
+            </TurnOnAdsButton>
+          : null
+        }
+        {
+          !showEnableAds
+          ? <AmountInformation data-test-id={`widget-amount-total-${type}`}>
+              <Amount>{amount}</Amount>
+              <ConvertedAmount>
+                {`${getLocale('bat')} (${converted} USD)`}
+              </ConvertedAmount>
+            </AmountInformation>
+          : null
+        }
+        <AmountDescription>
+          {
+            type === 'ads'
+            ? getLocale('rewardsWidgetEstimatedEarnings')
+            : getLocale('rewardsWidgetMonthlyTips')
+          }
+        </AmountDescription>
+      </AmountItem>
+    )
+  }
+
+  renderRewardsInfo = () => {
+    const {
+      enabledMain,
+      walletCreated
+    } = this.props
+
+    if (!enabledMain || !walletCreated) {
+      return null
+    }
+
+    return (
+      <div data-test-id2={'enableMain'}>
+        {this.renderAmountItem('ads')}
```

Comment:
can you change this from string to enum, so that we can't avoid any potential misspells